### DISCUSSION TITLE
#[graphql(skip)]   and  #[graphql(default_with = "date_now()")] problem

### DISCUSSION THREAD

**Discussion by gabriiels** [async-graphql/async-graphql#1455] [Q&A]
Hello!

How do you set a default date in an input? In the example I provide below, the 'skip' is activated and as a result, it sets the date to '1970-01-01T00:00:00+00:00' and does not call the function 'date_now()'.
I see that 'skip' and 'default_with' are not compatible with each other.
How can I make it so that it is skipped and has a default value that I want?
Thank you very much.

```
#[derive(InputObject, Deserialize, Serialize, Clone, Debug)]
pub struct CategoryInput {
    pub name: String,
    pub type_category: TypeCategoryInput,
    #[graphql(skip)]
    #[graphql(default_with = "date_now()")]
    pub created_at: DateTime<FixedOffset>,
    #[graphql(skip)]
    #[graphql(default_with = "date_now()")]
    pub update_at: DateTime<FixedOffset>,
}

#[derive(InputObject, Deserialize, Serialize, Clone, Debug)]
pub struct TypeCategoryInput {
    pub name: String,
    #[graphql(skip)]
    #[graphql(default_with = "date_now()")]
    pub created_at: DateTime<FixedOffset>,
    #[graphql(skip)]
    #[graphql(default_with = "date_now()")]
    pub update_at: DateTime<FixedOffset>,
}
```

**Comment by Miaxos** [comment]
Well, right now you cannot, the skip[^1] is before the default value[^2].

For now you can put it inside a new type with a custom impl of Default with date_now().

Why do you skip it and want a default value? Why don't you remove it and use `date_now` when you need it?

[^1]: https://github.com/async-graphql/async-graphql/blob/9297021ec4b29342cc75eb0d13a0b3c458a75a32/derive/src/input_object.rs#L74-L80
[^2]: https://github.com/async-graphql/async-graphql/blob/9297021ec4b29342cc75eb0d13a0b3c458a75a32/derive/src/input_object.rs#L133
